### PR TITLE
Simplify `Tap#remote_repo`.

### PR DESCRIPTION
**Pull request by reitermarkus**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

Extracted from https://github.com/Homebrew/brew/pull/16730 for easier review.

### CODE REVIEW

**Comment by apainintheneck** [comment]
I'm not sure this is really simpler. It's a few less lines I guess.

**Comment by MikeMcQuaid** [comment]
Agreed with @apainintheneck, this is worse as it actually introduces more branching logic despite being fewer lines. Passing on this, sorry.

**Comment by reitermarkus** [comment]
It's needed for Sorbet to work. Note the signature is added in a later commit in the original PR.

**Comment by reitermarkus** [comment]
Feel free to suggest an alternative that only calls `remote` once.

**Comment by MikeMcQuaid** [comment]
> Feel free to suggest an alternative that only calls `remote` once.

`remote_repo = remote` and then use that variable instead.

**Comment by apainintheneck** [comment]
> It's needed to work with the signature of `Tap#remote`, which is added in a later commit in the original PR.

In that case, I'm okay with whatever appeases the type system. That goes for other PR too.

**Comment by apainintheneck** [comment]
@reitermarkus Maybe this shouldn't be a separate PR in that case though. It'd be easier to understand in context. That being said I appreciate you breaking up these in general PRs so they're easier to review.

File: Library/Homebrew/tap.rb [hunk 1]
```diff
@@ -196,12 +196,11 @@ def remote
...
+  sig { returns(T.nilable(String)) }
   def remote_repo
-    return unless remote
-
-    @remote_repo ||= remote.delete_prefix("https://github.com/")
-                           .delete_prefix("git@github.com:")
-                           .delete_suffix(".git")
+    @remote_repo ||= remote&.delete_prefix("https://github.com/")
+                           &.delete_prefix("git@github.com:")
+                           &.delete_suffix(".git")
```
Comment (MikeMcQuaid) [inline Library/Homebrew/tap.rb:203]:
```suggestion
    return unless (remote_repo = remote)

    @remote_repo ||= remote_repo.delete_prefix("https://github.com/")
                                .delete_prefix("git@github.com:")
                                .delete_suffix(".git")
```

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks @reitermarkus!

**Pull request merged**